### PR TITLE
Update to Gradle 6.8 and clean up

### PR DESCRIPTION
**Pull request by Romitou**
### Description
This Pull Request brings major changes to the build system. It updates the Gradle wrapper from 5.0 to 6.7 and reorganises the `build.gradle` file to make it more understandable, clear and concise. As indicated in an issue (#3084), I have also removed the publishing system. Some variables have been added to the `gradle.properties`, and remove the use of *GStrings* as much as possible. I also modified and improved the part of the tests creation code. These changes also correct the duplication of aliases (#3727). I have also replaced LocalDate with LocalTime, which _should_ resolve issue #3571. Feel free to tell me what I need to change!

This Pull Request also provides version compatibility between JDK 8 and JDK 15 when PR #3705 will be merged.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** #3084, #3571, #3727, #2869

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -153,60 +113,17 @@ license {
...
-compileTestJava.options.encoding = 'UTF-8'
-
 // Task to check that test scripts are named correctly
 tasks.create('testNaming') {
 	doLast {
 		// Regression tests
 		for (def file : project.file('src/test/skript/tests/regressions').listFiles()) {
 			def name = file.getName()
-			if (!name.toLowerCase().equals(name)) {
+			if (name.toLowerCase() != name) {
```
Comment (FranKusmiruk) [inline build.gradle:122]: why was this changed?

Comment (Romitou) [reply build.gradle:122]: in reality, i preferred to use this operator because i find it clearer. according to the [Groovy documentation](https://docs.groovy-lang.org/latest/html/documentation/core-operators.html#_identity_operator), it is also calling `equals`. but if you want me to revert this change, let me know!

Comment (FranKusmiruk) [reply build.gradle:122]: it's fine, I just forgot it calls equals in groovy

**Review by FranKusmiruk — APPROVED** [review]
